src/main/kotlin/ink/abb/pogo/scraper/tasks/EvolvePokemon.kt

```diff
@@ -0,0 +1,49 @@
+/**
+ * Pokemon Go Bot  Copyright (C) 2016  PokemonGoBot-authors (see authors.md for more information)
+ * This program comes with ABSOLUTELY NO WARRANTY;
+ * This is free software, and you are welcome to redistribute it under certain conditions.
+ *
+ * For more information, refer to the LICENSE file in this repositories root directory
+ */
+
+package ink.abb.pogo.scraper.tasks
+
+import ink.abb.pogo.scraper.Bot
+import ink.abb.pogo.scraper.Context
+import ink.abb.pogo.scraper.Settings
+import ink.abb.pogo.scraper.Task
+import ink.abb.pogo.scraper.util.Log
+
+/**
+ * @author Michael Meehan (Javapt)
+ */
+
+class  EvolvePokemon : Task {
+    override fun run(bot: Bot, ctx: Context, settings: Settings) {
+        if (settings.autoEvolve.isEmpty()) {
+            return
+        }
+        val groupedPokemon = ctx.api.inventories.pokebank.pokemons.groupBy { it.pokemonId }
+        val autoEvolve = settings.autoEvolve
+        val canEvolve = groupedPokemon.filter {
+            val candyNeeded = settings.candyRequiredByPokemon[it.key.number]
+            candyNeeded != null && candyNeeded > 0 && autoEvolve.contains(it.key.name) && it.value.first().candy >= candyNeeded
+        }
+        if (canEvolve.isEmpty()) {
+            return
+        }
+        canEvolve.forEach {
+            val sorted = it.value.sortedByDescending { it.cp }

```

Comment:
Yea just change this: val sorted = it.value.sortedByDescending { it.cp } with this: 
            val sorted = if (settings.sortByIV) {
                it.value.sortedByDescending { it.getIv() }
            } else {
                it.value.sortedByDescending { it.cp }
            }
I dont know how to make a pull request so i only can write it here sorry!